pkgs/applications/graphics/krita/default.nix

```diff
@@ -44,6 +44,12 @@ mkDerivation rec {
     "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
   ];
 
+  postInstall = ''
+    for i in $out/bin/*; do
+      wrapProgram $i --prefix PYTHONPATH : "$PYTHONPATH"
```

Comment:
Can't we use `python3.pkgs.wrapPython`?

Reply:
I'm slightly confused on how it is supposed to be used. (syntax wise)
It should probably be used like this:
wrapPythonProgramsIn "$out/bin" "$out $buildInputs"
right (though I can probably leave out the second $out)?

Taking a look at [mk-python-derivation.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/mk-python-derivation.nix) which is used by [buildPythonPackage](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/build-python-package.nix) I'm not quite sure where pythonPath normally comes from. (it seems to be deprecated and should be in most cases empty (and is otherwise added to propagatedBuildInputs) the wrapper calls wrapPythonPrograms in it's postFixup phase nonetheless...)

Otherwise this seems to be exactly what I've been looking for.

Reply:
My understanding is that this wraps python scripts, by default in `$out/bin`... but krita has no scripts in there.  It would perhaps be possible and cleaner to find all/relevant scripts in the output and patch them like this, but I expect the approach of this PR to be easier.

Reply:
`buildPythonPath "$out $buildInputs"` generates a cleaned up `program_PYTHONPATH` variable containing the `PYTHONPATH`.
```
/nix/store/sm0x0zxv583a73hlcbj8rg3famla552a-python3.7-setuptools-41.0.1/lib/python3.7/site-packages
/nix/store/r0zzxla0nbwp3mfn4j0gckz2llkyza2m-python3.7-dbus-python-1.2.4/lib/python3.7/site-packages
/nix/store/07lmhxfm62kqdh38w2j5skv27rfqpg5x-python3.7-PyQt-5.11.3/lib/python3.7/site-packages
```
Because sip is missing in krita's and pyqt5's buildInputs and propagatedBuildInputs respectively it ends up not being added. (and krita once again complains about it missing)
For some reason setuptools is included though.

Reply:
I would like some feedback on whether I should keep it as it is or change it to use buildPythonPath.
Also, once everything is done I should I [squash my commits](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches)?

In any case, I really appreciate the amount of feedback I've gotten so far.  Thanks a lot!